kind/kind-up-test.sh

```diff
@@ -0,0 +1,46 @@
+# run this from root of repository
+
+# spin up kind cluster
+cat <<EOF | sudo kind create cluster --image kindest/node:v1.28.0 --config=-
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  kubeadmConfigPatches:
+  - |
+    kind: InitConfiguration
+    nodeRegistration:
+      kubeletExtraArgs:
+        node-labels: "ingress-ready=true"
+  extraPortMappings:
+  - containerPort: 80
+    hostPort: 80
+    protocol: TCP
+  - containerPort: 443
+    hostPort: 443
+    protocol: TCP
+EOF
+
+sudo kind get kubeconfig > /tmp/config
+sudo chown $USER:$USER /tmp/config
+mv /tmp/config ~/.kube/config
+
+# install ingress-nginx
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/main/deploy/static/provider/kind/deploy.yaml
+
+#kubectl wait --namespace ingress-nginx \
+#  --for=condition=ready pod \
+#  --selector=app.kubernetes.io/component=controller \
+#  --timeout=90s
+#
+# TODO: add a wait for ingress to be ready with test.yaml & curl
+#sleep 20
+
+oc create ns fulcio-system
+oc create ns rekor-system
+oc -n fulcio-system create secret generic fulcio-secret-rh --from-file=private=./kind/test-keys-cert/file_ca_key.pem --from-file=public=./kind/test-keys-cert/file_ca_pub.pem --from-file=cert=./kind/test-keys-cert/fulcio-root.pem  --from-literal=password=secure --dry-run=client -o yaml | oc apply -f-
```

Comment:
How were the files in this directory (`./kind/test-keys-cert/`) and the directory itself generated? I dont see them provided in the repo and I am not sure how to reproduce them.

Reply:
I must have added a gitignore for those - I think we can push them & use/re-use them since they are just for testing (unless we can't/shouldn't) ... I'll push them just a sec

Reply:
also I added a rh-gitleaks-precommit so we don't get bugged about leaking sensitive data